pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -98,6 +98,11 @@ func (s *FlinkStateMachine) shouldRollback(ctx context.Context, application *v1a
 
 func (s *FlinkStateMachine) Handle(ctx context.Context, application *v1alpha1.FlinkApplication) error {
 	currentPhase := application.Status.Phase
+	if _, ok := s.metrics.stateMachineHandlePhaseMap[currentPhase]; !ok {
+		errMsg := fmt.Sprintf("Invalid state %s for the application", currentPhase)
+		logger.Errorf(ctx, errMsg)
```

Comment:
Do we need to explicitly log this here? Looks like the error returned from Handle is already logged in Reconcile.

Reply:
oh sure. I can remove it.